### PR TITLE
 [5.7] Container bindMethod improvement

### PR DESCRIPTION
**Pull request by TBlindaruk**
resubmit of https://github.com/laravel/framework/pull/23767

### CODE REVIEW

File: src/Illuminate/Container/Container.php [hunk 1]
```diff
@@ -285,11 +286,17 @@ public function bindMethod($method, $callback)
...
      * @param  array|string $method
      * @return string
+     *
+     * @throws \InvalidArgumentException
      */
     protected function parseBindMethod($method)
     {
         if (is_array($method)) {
-            return $method[0].'@'.$method[1];
+            if (($count = count($method)) > 1) {
```
Comment (a-komarev) [inline src/Illuminate/Container/Container.php]: I think its better to throw exception in conditional, and place return after it.

```php
if (is_array($method)) {
    if (($count = count($method)) < 2) {
        throw new InvalidArgumentException("Method should be array with length >= 2, array with length [$count] given");
    }

    return $method[0].'@'.$method[1];
}
```

Comment (TBlindaruk) [reply src/Illuminate/Container/Container.php]: @a-komarev , changed

File: src/Illuminate/Container/Container.php [hunk 2]
```diff
@@ -285,10 +286,16 @@ public function bindMethod($method, $callback)
      *
      * @param  array|string $method
      * @return string
+     *
+     * @throws \InvalidArgumentException
      */
     protected function parseBindMethod($method)
     {
         if (is_array($method)) {
```
Comment (phanan) [inline src/Illuminate/Container/Container.php]: I'd use early returns here to avoid the awkward nested `if`s.

Comment (TBlindaruk) [reply src/Illuminate/Container/Container.php]: I had idea to write like 
```PHP
    protected function parseBindMethod($method)
    {
        if (is_array($method) && ($count = count($method) < 2)) {
            throw new InvalidArgumentException("Method should be array with length >= 2, array with length [{$count}] given");
        }
    
        if (is_array($method)) {
            return $method[0] . '@' . $method[1];
        }

        return $method;
    }
```

But I`m not sure... is it really better?

**Comment by taylorotwell** [comment]
Ton of other changes here.

**Comment by TBlindaruk** [comment]
@taylorotwell , @GrahamCampbell ,

It happened after 
`@GrahamCampbell GrahamCampbell changed the base branch from 5.6  to master  36 minutes ago`

Should I create PR to master brunch?

**Pull request closed without merging**